website/content/en/docs/tasks/deprov-nodes.md

```diff
@@ -1,36 +1,84 @@
 ---
 
+Karpenter sets a Kubernetes [finalizer](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/) on each node it provisions.
+The finalizer specifies additional actions the Karpenter controller will takes in response to a node deletion request.
+These include:
 
-## Deletion Workflow
+* Marking the node as unschedulable, so no further pods can be deployed there.
+* Evicting all workload pods from the node, draining it of all pods other than daemonsets.
+* Deleting the node from the Kubernetes cluster.
+* Terminating the instance from the cloud provider.
 
-### Finalizer
+## How Karpenter nodes are deprovisioned
 
-Karpenter adds a finalizer to provisioned nodes. [Review how finalizers work.](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/#how-finalizers-work) 
+There are both automated and manual ways of deprovisioning nodes provisioned by Karpenter:
 
-### Drain Nodes
+* **Node empty**: Karpenter notes when the last workload (non-daemonset) pod stops running on a node. From that point, Karpenter waits the number of seconds set by `ttlSecondsAfterEmpty`  in the provisioner, then Karpenter requests to delete the node. This feature can keep costs down by removing nodes that are no longer being used for workloads.
+* **Node expired**: Karpenter asks to delete the node after a set number of seconds, based on the provisioner `ttlSecondsUntilExpired`  value, from the time the node was provisioned. One use case for node expiry is to handle node upgrades. Old nodes (with a potentially outdated Kubernetes version or operating system) are deleted, and replaced with nodes on the current version.
 
-Review how to [safely drain a node](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/).
+    {{% alert title="Note" color="primary" %}}
+    Keep in mind that a small NodeExpiry results in a higher churn in cluster activity. So, for example, if a cluster brings up all nodes at once, all the pods on those nodes would fall into the same batching window on expiration.
+    {{% /alert %}}
+    
+* **Node deleted**: You could use `kubectl` to remove a single Karpenter node:
 
-## Delete Node
+    ```bash
+    kubectl delete <nodename>
+    ```
 
-Karpenter changes the behavior of `kubectl delete node`. Nodes will be drained, and then the underlying instance will be deleted.
+    Or delete all Karpenter nodes at once:
 
-## Disruption Budget
+    ```bash
+    kubectl delete -l <provisioner label>
+ 
+    ```
 
-Karpenter respects Pod Disruption Budgets. Review what [disruptions are](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/), and [how to configure them](https://kubernetes.io/docs/tasks/run-application/configure-pdb/).
+Whether through node expiry or manual deletion, Karpenter seeks to follow graceful termination procedures as described in Kubernetes [Graceful node shutdown](https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdow) documentation.
```

Comment:
Do we want to add something about how if the karpenter controller is removed (or fails), the finalizers on the nodes will be orphaned, and will need manual removal?

Reply:
Done.